src/components/navbars/DesktopNav.jsx

```diff
@@ -1,32 +1,39 @@
+import ProfileNav from './ProfileNav';
 import loginIcon from '../../assets/icons/login.png';
 import searchIcon from '../../assets/icons/search.png';
+import { useAuth } from '../../AuthProvider';
 
 const DesktopNav = ({ onSearch, onLogin, onJoin, stateSearch, stateLogin, stateJoin }) => {
+    const { isLoggedIn } = useAuth()
     return (
-        <nav className="w-2/3 hidden relative py-6 sm:flex justify-between">
+        <nav className="w-2/3 hidden relative py-6 sm:flex justify-between items-center">
             {
                 stateSearch
                     ? <form className="border-2 border-red rounded-full h-10 w-2/4 flex justify-between items-center px-4 ml-6 gap-1">
-                        <input className='focus:outline-none w-2/3' aria-label="Search" placeholder="Search online classes or teachers..." />
+                        <input className='focus:outline-none w-2/3' aria-label="Search" placeholder="Search classes or teachers..." />
                         <button onClick={onSearch} aria-label="Search"><img src={searchIcon} className="h-6" alt="search icon" /></button>
                     </form>
                     : <div></div>
             }
             <div id="mainNav" className="flex gap-4 items-center">
-                {
-                    stateLogin
-                        ? <button onClick={onLogin} className="font-spartan font-semibold text-base sm:text-lg flex items-center">
-                            <img src={loginIcon} className="h-4 sm:h-5" alt="join icon" />
-                            Log In
-                        </button>
-                        : <div></div>
-                }
-                {
-                    stateJoin
-                        ? <button onClick={onJoin} className='bg-red hover:bg-pureWhite hover:text-red w-16 h-8 hover:border-2 hover:border-red text-white font-spartan font-semibold text-base sm:text-lg rounded-lg transition duration-300 easy-in'>
-                            Join
-                        </button>
-                        : <div></div>
+                {isLoggedIn
+                    ? <ProfileNav />
+                    : <>
+                        {stateLogin
+                            ? <button onClick={onLogin} className="font-spartan font-semibold text-base sm:text-lg flex items-center">
+                                <img src={loginIcon} className="h-4 sm:h-5" alt="join icon" />
+                                Log In
+                            </button>
+                            : <div></div>
```

Comment:
This could be `button` as It'll provided some context as to what is missing but if you intend for nothing to be rendered and there is no need to provide any context you can use a conditional expression with logical AND. That'll will only render the component if the state is `true`.  

I noticed you've used that in other parts of your code so if the purpose of the empty tags is to provide a place holder to help with styling, that should be fine for now. But if you have time try and remove them.